o2a/transformers/base_transformer.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base transformer classes"""
+from abc import ABC, abstractmethod
+from typing import Type
+
+from o2a.converter.parsed_action_node import ParsedActionNode
+from o2a.converter.workflow import Workflow
+
+
+# pylint: disable=too-few-public-methods
+class BaseWorkflowTransformer(ABC):
+    """
+    Base class for all transformers
+    """
+
+    @abstractmethod
+    def process_workflow(self, workflow: Workflow):
```

Comment:
```suggestion
    def   def __call__(self, workflow: Workflow):
```
I personally would like to use this here, but I do not know if it will be legible enought for other people. This will also make pylint happy

Reply:
I think we should stay with process_workflow but join the two classes TypeNodeWorkflowTransformer and BaseWorkflowTransformer into single BaseWorkflowTransformer (see below).

Reply:
My change suggests that the transformers should be function. In many cases, it does not need a class. In other cases, we can create a class using the __call__ method

Reply:
Yeah. But that is pretty obscure use. I would prefer to keep classes - together with type annotations I think it's a bit cleaner this way (and more structured). Also this makes it better for testing -we can test classes rather than functions. Reading is more important than writing and having __call__ adds implicit behaviour which is not that obvious. I think we should stick to plain object/classe structure.

Reply:
I updated the PR. We have only base class now.